lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -7,25 +7,21 @@ define((require, exports, module) ->
         placement: "bottom"
         trigger: "manual"
         title: "Formula"
-        container: "#editor"
+        container: editor.container
       }
 
-      show: (jqPopoverContainer, content) ->
-        cursorPosition = $("textarea.ace_text-input").position()
-        jqPopoverContainer.css({
-          top: "#{cursorPosition.top + 24}px"
-          left: "#{cursorPosition.left}px"
-        })
+      show: (jqPopoverContainer, content, position) ->
+        jqPopoverContainer.css(position)
         popoverHandler.options.content = content
         jqPopoverContainer.popover(popoverHandler.options)
         jqPopoverContainer.popover("show")
         return
 
-      hide: (jqPopoverContainer) ->
+      destroy: (jqPopoverContainer) ->
         jqPopoverContainer.popover("destroy")
 
-      isVisible: (jqPopoverElement) ->
-        jqPopoverElement.children(".popover").is(":visible")
+      isVisible: (jqPopoverContainer) ->
+        jqPopoverContainer.data().popover.tip().hasClass("in")
```

Comment:
Yes, I know that Bootstrap 3 and 4 change API in incompatible ways and there are many other issues besides popovers. However, the rule still holds in general case: long chain of calls is fragile and error prone. 

Why do you want the tip so much ;) ? This method is supposed to return `true` if popover is visible, the most straightforward way of checking is to check whether the container of all popover's elements is visible. You suggest checking if one small element from the container is visible. Why that element, why not popover-content or popover-title? Will it work in imaginary case when popover has no arrow? You leave so many questions to the reader of this code.